File: backend/dto/friend.py
from pydantic import BaseModel
from dto import DB_User
import uuid

# /api/friends
# GET
class GetFriendResponse(BaseModel):
    users: list[DB_User] | None = None

class GetFriendRequestsResponse(BaseModel):
    users: list[DB_User] | None = None

# POST
class PostFriendRequest(BaseModel):
    requestor_id: uuid.UUID
    requestee_id: uuid.UUID

class PostFriendResponse(BaseModel):
    success: bool

File: frontend/src/dto/friend.ts
export interface User {
    user_id: string | null;
    username: string | null;
    online: boolean | null;
}

export interface GetFriendResponse {
    friend_ids: Array<string> | null
}

export interface GetFriendRequestsResponse {
    users: Array<User>
}

export interface PostFriendRequest {
    requestor_id: string,
    requestee_id: string,
}

export interface PostFriendResponse {
    success: boolean
}